azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfigurationTest.java

```diff
@@ -9,158 +9,76 @@
     }
 
     @Test
     public void cannotAutowireTopicClientWithInvalidConnectionString() {
-        System.setProperty(Constants.CONNECTION_STRING_PROPERTY, Constants.INVALID_CONNECTION_STRING);
-        System.setProperty(Constants.TOPIC_NAME_PROPERTY, Constants.TOPIC_NAME);
-
-        verifyBeanCreationException("Failed to instantiate [com.microsoft.azure.servicebus.TopicClient]: " +
-                "Factory method 'topicClient' threw exception; nested exception is " +
-                "com.microsoft.azure.servicebus.primitives.IllegalConnectionStringFormatException: " +
-                "Connection String cannot be parsed.", TopicClient.class);
+        this.contextRunner.withPropertyValues(Constants.CONNECTION_STRING_PROPERTY, Constants.INVALID_CONNECTION_STRING)
+                .withPropertyValues(Constants.TOPIC_NAME_PROPERTY, Constants.TOPIC_NAME);
 
-        System.clearProperty(Constants.CONNECTION_STRING_PROPERTY);
-        System.clearProperty(Constants.TOPIC_NAME_PROPERTY);
+        this.contextRunner.run((context) -> assertThat(context).doesNotHaveBean(TopicClient.class));
     }
 
     @Test
     public void cannotAutowireSubscriptionClientWithInvalidConnectionString() {
-        System.setProperty(Constants.CONNECTION_STRING_PROPERTY, Constants.INVALID_CONNECTION_STRING);
-        System.setProperty(Constants.TOPIC_NAME_PROPERTY, Constants.TOPIC_NAME);
-        System.setProperty(Constants.SUBSCRIPTION_NAME_PROPERTY, Constants.SUBSCRIPTION_NAME);
-        System.setProperty(Constants.SUBSCRIPTION_RECEIVE_MODE_PROPERTY, Constants.SUBSCRIPTION_RECEIVE_MODE.name());
+        this.contextRunner.withPropertyValues(Constants.CONNECTION_STRING_PROPERTY, Constants.INVALID_CONNECTION_STRING)
+                .withPropertyValues(Constants.TOPIC_NAME_PROPERTY, Constants.TOPIC_NAME)
+                .withPropertyValues(Constants.SUBSCRIPTION_NAME_PROPERTY, Constants.SUBSCRIPTION_NAME)
+                .withPropertyValues(Constants.SUBSCRIPTION_RECEIVE_MODE_PROPERTY,
+                        Constants.SUBSCRIPTION_RECEIVE_MODE.name());
 
-        verifyBeanCreationException("Failed to instantiate [com.microsoft.azure.servicebus.SubscriptionClient]: " +
-                "Factory method 'subscriptionClient' threw exception; nested exception is " +
-                "com.microsoft.azure.servicebus.primitives.IllegalConnectionStringFormatException: " +
-                "Connection String cannot be parsed.", SubscriptionClient.class);
-
-        System.clearProperty(Constants.CONNECTION_STRING_PROPERTY);
-        System.clearProperty(Constants.TOPIC_NAME_PROPERTY);
-        System.clearProperty(Constants.SUBSCRIPTION_NAME_PROPERTY);
-        System.clearProperty(Constants.SUBSCRIPTION_RECEIVE_MODE_PROPERTY);
+        this.contextRunner.run((context) -> assertThat(context).doesNotHaveBean(SubscriptionClient.class));
     }
 
     @Test
     public void cannotAutowireSubscriptionClientWithInvalidCredential() {
```

Comment:
I suppose invalid credential should throw exception. It should be asserted also.